recipe-client-addon/lib/NormandyApi.jsm

```diff
@@ -145,20 +149,65 @@ this.NormandyApi = {
 
   /**
    * Fetch an array of available actions from the server.
+   * @param filters
+   * @param filters.enabled {boolean} If true, only returns enabled
+   * recipes. Default true.
+   * @resolves {Array}
+   */
+  async fetchRecipes(filters = {enabled: true}) {
+    return this.fetchSignedObjects('recipe', filters);
+  },
+
+  /**
+   * Fetch an array of available actions from the server.
    * @resolves {Array}
    */
-  async fetchActions() {
-    const actionApiUrl = await this.getApiUrl("action-list");
-    const res = await this.get(actionApiUrl);
-    return res.json();
+  async fetchActions(filters = {}) {
+    return this.fetchSignedObjects('action', filters);
   },
 
   async fetchImplementation(action) {
-    const response = await fetch(action.implementation_url);
-    if (response.ok) {
-      return response.text();
+    const implementationUrl = this.absolutify(action.implementation_url);
```

Comment:
Our public APIs tend to, but for various reasons our dev servers don't always. We have this utility already, so it probably works fine to keep it.